### PR TITLE
Convert step-by-step guide to markdown

### PR DESCRIPTION
**Pull request by forsyth2**
Converting the [original step-by-step guide](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/2309226536/Running+E3SM+step-by-step+guide) from Confluence to Markdown.

### CODE REVIEW

**Comment by forsyth2** [comment]
# Primary updates
Most obviously, the guide has been converted from Confluence to Markdown. Other differences:
- The guide now refers to the v3 template script.
- Cori references are replaced with Perlmutter.
- I removed references to code older than Unified 1.6.0, since that is significantly out of date by now.
- Similarly, I removed a warning about https://github.com/E3SM-Project/zstash/issues/167 since that has been resolved.
- Misc. readability improvements

The full output can be seen at https://github.com/forsyth2/E3SM/blob/forsyth2/docs/update-guide/docs/step_by_step_guide.md

# Documentation building
I followed the directions at [Developing Documentation](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/3924787306/Developing+Documentation), but I ended up getting:
```
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/mkdocs/__main__.py", line 192, in build_command
    build.build(config.load_config(**kwargs), dirty=not clean)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/mkdocs/commands/build.py", line 257, in build
    config = config['plugins'].run_event('config', config)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/mkdocs/plugins.py", line 102, in run_event
    result = method(item, **kwargs)
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/mkdocs_monorepo_plugin/plugin.py", line 52, in on_config
    new_docs_dir = self.merger.merge()
  File "/gpfs/fs1/home/ac.forsyth2/mkdocsenv/lib64/python3.6/site-packages/mkdocs_monorepo_plugin/merger.py", line 62, in merge
    copytree(source_dir, dest_dir, dirs_exist_ok=True)
TypeError: copytree() got an unexpected keyword argument 'dirs_exist_ok'
```
I even set up a new clone of E3SM to work in, and followed the relevant directions on the [Development Getting Started Guide](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/1868455/Development+Getting+Started+Guide).

I wasn't able to build the docs before _or_ after I made the changes here.

# Other notes
I was trying to see if Confluence-to-Markdown exporters exist, but it looked like that could only be accomplished using non-Atlassian plugins. E.g., [here](https://community.atlassian.com/t5/Confluence-questions/Does-anyone-know-of-a-way-to-export-from-confluence-to-MarkDown/qaq-p/921126) and [here](https://community.atlassian.com/t5/Confluence-questions/export-to-markdown/qaq-p/307481). [Markdown Exporter for Confluence](https://marketplace.atlassian.com/apps/1221351/markdown-exporter-for-confluence?tab=overview&hosting=cloud) is such an app. 

But I ultimately decided to just port the guide over manually for two reasons:
1. There were a number of changes that needed to be made, so it gave me a chance to go through the guide line-by-line.
2. I wasn't sure about the security constraints of our Confluence space -- e.g., if we needed to get apps approved first. For example, [Installing Marketplace apps](https://confluence.atlassian.com/upm/installing-add-ons-273875715.html) mentions "A note about safety: apps are very powerful, they can change the behavior of almost any part of the Atlassian application. It's very important that you trust an app before you install it. Always be aware of where (and who) an app comes from."

File: docs/step_by_step_guide.md [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+- `readonly MACHINE=pm-cpu`: the name of the machine you’re running on.
+- `readonly PROJECT="e3sm"`: SLURM project accounting (typically `e3sm`).
+
+`# Simulation`
+- `readonly COMPSET="WCYCL1850"` : compset (configuration)
+- `readonly RESOLUTION="ne30pg2_r05_IcoswISC30E3r5"`: resolution
+  - `ne30` is the number of spectral elements for the atmospheric dynamics grid
+  - `pg2` refers to the physics grid option. This mesh grid spacing is approximately 110 km.
+  - `r05` is the ... <!-- (TODO: define) -->
+  - `IcoswISC30E3r5` is the ... <!-- (TODO: define) -->
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:57]: What do these parts of the resolution refer to?

Alternatively, I can skip these sub-bullets altogether, but I think they're useful for understanding the naming conventions.

File: docs/step_by_step_guide.md [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+## Output
+The post-processing output is organized hierarchically. Examples:
+- `<e3sm_simulations_dir>/<case_name>/post/atm/180x360_aave/ts/monthly/10yr` has the time series files – one variable per file, in 10 year periods as defined in `<run_scripts_dir>/post.<case_name>.cfg`.  
+- `<e3sm_simulations_dir>/<case_name>/post/atm/180x360_aave/clim/20yr` similarly has climatology files for 20 year periods, as defined in `<run_scripts_dir>/post.<case_name>.cfg``.
+- `<e3sm_simulations_dir>/<case_name>/post/atm/glb/ts/monthly/10yr` has globally averaged files for 10 years periods as defined in `<run_scripts_dir>/post.<case_name>.cfg`.
+
+# Documenting the Model Run
+You should create a Confluence page for your model run in the relevant Confluence space. Use the [Simulation Run Template](https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/2297299190) as a template. See below for how to fill out this template.
+
+<!-- TODO: where should the Confluence pages be made for v3 (and for each group)? -->
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:365]: Is there a newer template that should be used?  What Confluence directory should v3 simulations be documented in?

File: docs/step_by_step_guide.md [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+
+## MPAS Analysis
+
+> [!NOTE]
+> The plots will be available online at the URL corresponding to `<www>/mpas_analysis/` (where `www` is specified in the `zppy` cfg). See the [E3SM Diags quick guide](https://e3sm-project.github.io/e3sm_diags/_build/html/master/quickguides/quick-guide-general.html) to find the URLs for the web portals on each E3SM machine (listed as `<web_address>`).
+
+Make a bulleted list of links, e.g., for `<url_path>/ts_0001-0050_climo_0021-0050/`, create a bullet `“1-50 (time series), 21-50 (climatology)”.
+
+## ILAMB
+<!-- TODO: Add this section? -->
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:438]: The original guide was written with Water Cycle in mind. Is there a preferred way to include ILAMB links in simulation documentation?

File: docs/step_by_step_guide.md [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+$ ls /home/<first letter>/<username>/E3SMv2/<case_name>
+# tar files, `index.db`, `zstash_create` log, and `zstash_check` log should be present
+$ exit
+```
+Update the simulation Confluence page with information regarding this simulation (For Water Cycle’s v2 work, that page is [V2 Simulation Planning](https://acme-climate.atlassian.net/wiki/spaces/ED/pages/2766340117)). In the `zstash archive` column, specify: 
+- `/home/<first letter>/<username>/E3SMv2/<case_name>`
+- `zstash_create_<stamp>.log`
+- `zstash_check_<stamp>.log`
+
+<!-- TODO: Is there a v3 planning page to link? -->
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:616]: Is there a corresponding v3 simulation planning page with information about all the simulations?

**Review by forsyth2 — COMMENTED** [review]
I've left some questions re: the best ways to update this guide.

**Comment by forsyth2** [comment]
@mccoy20 and @chengzhuzhang requested a thorough update to the step-by-step guide, since it was written a while ago. @golaz @chengzhuzhang requested that the new version be done in Markdown rather than Confluence. I have implemented these requests here.

@golaz, @wlin7, @ndkeen -- @chengzhuzhang suggested you three as reviewers to check if the new guide is properly up-to-date. I'm therefore marking you all as reviewers. Thank you.

@rljacob, @wlin7 -- @chengzhuzhang suggested you two as being able to provide more instruction on including the Markdown guide into the E3SM Documentation. Thank you. I've followed the directions at [Developing Documentation](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/3924787306/Developing+Documentation), but I wasn't able to get the documentation built (see note above). I'm also not quite sure how to link in the new guide. That page mentions a `user-guide` directory, but there is no such directory under `E3SM/docs`. Furthermore, it looks like most of the documentation covers the individual components (i.e., the [index](https://github.com/E3SM-Project/E3SM/blob/forsyth2/docs/update-guide/docs/index.md?plain=1) only links to components).

**Comment by forsyth2** [comment]
> Could you try with the latest e3sm-unified (1.9.3)?

Ah, I didn't realize just using Unified would work. With that, I can start serving the documentation, but I now realize I have no way of looking at it, since it's serving locally on Chrysalis (where I do the bulk of my development) and not on my own machine. I figured just using the GitHub markdown rendering would work fine since it's just one page (as opposed to many connected ones).

> could you please link this new file so that it appears in the docs?

The second commit includes this change.

File: docs/step_by_step_guide.md [hunk 5]
```diff
@@ -0,0 +1,646 @@
+# Useful Aliases
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:1]: Adding a section above "Useful Aliases"

```
# About This Guide
This guide is intended to walk users step-by-step through the simulation process -- from setting up a simulation to post-processing, long-term archiving, and publishing it. 
```

File: docs/step_by_step_guide.md [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+Setting some aliases may be useful in running the model. You can edit your bash settings file to add aliases. Run `source` on that file to start using your new aliases. Examples:
+- Chrysalis – `source ~/.bashrc`
+- Compy – `source ~/.bash_profile`
+- Perlmutter – `source ~/.bash_profile.ext`
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:5]: These bullets show up fine on the GitHub rendering but all in one line on the actual rendering...

File: docs/step_by_step_guide.md [hunk 7]
```diff
@@ -0,0 +1,646 @@
...
+
+Note that the specific file name may differ amongst machines. For example, it might be named `~/.bashrc`.ext.
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:7]: `~/.bashrc`.ext. => `~/.bashrc.ext.`

File: docs/step_by_step_guide.md [hunk 8]
```diff
@@ -0,0 +1,646 @@
...
+The output of `sq` uses several abbreviations: ST = Status, R = running, PD = pending, CG = completing.
+
+## Directories
+You will be working in several directories. 
+
+<run_scripts_dir>: `${HOME}/E3SM/scripts`
+
+<code_source_dir>: `${HOME}/E3SM/code`
+
+<simulations_dir> differs amongst machines:
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:27]: These show up fine in the GitHub rendering, but in the actual rendering, anything in `<>` brackets simply shows up as an empty string...

File: docs/step_by_step_guide.md [hunk 9]
```diff
@@ -0,0 +1,646 @@
...
+```shell
+# Model running
+alias run_scripts="cd <run_scripts_dir>"
+alias simulations="cd <simulations_dir>"
+```
+
+# Configuring the Model Run – Run Script
+Start with an example of a run script for a low-resolution coupled simulation:  
+
+We'll use the template script in the E3SM repository, which uses Perlmutter: https://github.com/E3SM-Project/E3SM/blob/master/run_e3sm.template.sh
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:43]: This link works in the GitHub rendering, but not the actual rendering. In any case, I'm replacing it with this line:
`We'll use the [template script](https://github.com/E3SM-Project/E3SM/blob/master/run_e3sm.template.sh) in the E3SM repository, which uses Perlmutter.`

File: docs/step_by_step_guide.md (hunk 8 above)
Comment (forsyth2) [reply docs/step_by_step_guide.md:27]: This problem recurs throughout the file.

File: docs/step_by_step_guide.md (hunk 6 above)
Comment (forsyth2) [reply docs/step_by_step_guide.md:5]: This problem recurs throughout the file.

File: docs/step_by_step_guide.md [hunk 10]
```diff
@@ -0,0 +1,646 @@
...
+- `readonly COMPSET="WCYCL1850"`: compset (configuration)
+- `readonly RESOLUTION="ne30pg2_r05_IcoswISC30E3r5"`: resolution. In this example, we have:
+  - `ne30pg2`: atmosphere (ne30 dynamics grid -- 30 spectral elements, pg2 physics grid)
+  - `r05`: land and river on 1/2 lat/lon grid (commonly referred to as "tri-grid")
+  - `IcoswISC30E3r5`: ocean and sea-ice on Icosahedral 30 km mesh with ice shelves cavities (wISC), E3SMv3 (E3) revision r5.
+- `readonly CASE_NAME="your_casename"`: case name
+- `# readonly CASE_GROUP=""`: This will let you mark multiple cases as part of the same group for later processing (e.g., with PACE). 
+
+> [!IMPORTANT]
+> If this is part of a simulation campaign, ask your group lead about using a `CASE_GROUP` label. Otherwise, please use a unique name to distinguish from existing `CASE_GROUP` label names, (e.g., “v2.LR“).
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:61]: This note box works in the GitHub rendering, but not fully correct in the actual rendering. It does show up as a sidebar/quote block, but the information about the note type is lost (and it just prints the `[!IMPORTANT]` as if it were regular text).

Comment (forsyth2) [reply docs/step_by_step_guide.md:61]: This problem recurs throughout the file.

File: docs/step_by_step_guide.md [hunk 11]
```diff
@@ -0,0 +1,646 @@
...
+
+The first time the script is called, all the flags should be set to true. Subsequently, the user may decide to bypass code checkout (`do_fetch_code=false`) or compilation (`do_case_build=false`). A user may also prefer to manually submit the job by setting `do_case_submit=false` and then invoking `./case.submit`.
+
+# Running the Model
+
+## Short Tests
+Before starting a long production run, it is *highly recommended* to perform a few short tests to verify:
+1. The model starts without errors.
+2. The model produces BFB (bit-for-bit) results after a restart.
+3. The model produces BFB results when changing PE layout.
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:113]: The numbered list works in the GitHub rendering, but not in the actual rendering (where it shows up all in one line).

File: docs/step_by_step_guide.md [hunk 12]
```diff
@@ -0,0 +1,646 @@
...
+```
+Each component of the model has a directory under `archive/`. There are also two additional directories under `archive/`: `logs` holds the gzipped log files and `rest` holds the restart files.
+| Component | Directory | File naming pattern |
+| --- | --- | --- |
+| Atmosphere (Earth Atmospheric Model) | `archive/atm/hist` | `*.eam.h*` |
+| Coupler | `archive/cpl/hist` | `*.cpl.h*` |
+| Sea Ice (MPAS-Sea-Ice) | `archive/ice/hist` | `*.mpassi.hist.*` |
+| Land (Earth Land Model) | `archive/lnd/hist` | `*.elm.h*` |
+| Ocean (MPAS-Ocean) | `archive/ocn/hist` | `*.mpaso.hist.*` |
+| River Runoff (MOSART) | `archive/rof/hist` | `*.mosart.h*` |
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:238]: The table shows up in the GitHub rendering, but all as one line in the actual rendering...

File: docs/step_by_step_guide.md [hunk 13]
```diff
@@ -0,0 +1,646 @@
...
+# Short Term Archiving
+By default, E3SM will store all output files under the `<simulations_dir>/<case_name>/run/` directory. For long simulations, there could 10,000s to 100,000s of output files. Having so many files in a single directory can be very impractical, slowing down simple operations like `ls` to a crawl. CIME includes a short-term archiving utility that will neatly organize output files into a separate `<simulations_dir>/<case_name>/archive/` directory. Short term archiving can be accomplished with the following steps. 
+
+> [!TIP]
+> This can be done while the model is still running.
+
+Use `--force-move` to move instead of copying, which can take a long time. Set `--last-date` to the latest date in the simulation you want to archive. You do not have to specify a beginning date.
+```shell
+cd <simulations_dir>/<case_name>/case_scripts
+./case.st_archive --last-date 0051-01-01 --force-move --no-incomplete-logs
```
Comment (forsyth2) [inline docs/step_by_step_guide.md:227]: `0051` shows up as red in the actual rendering for some reason

**Review by forsyth2 — COMMENTED** [review]
Self-review using the [actual rendering](https://e3sm-project.github.io/E3SM/pr-preview/pr-6326/step_by_step_guide/) and the [GitHub rendering](https://github.com/E3SM-Project/E3SM/blob/forsyth2/docs/update-guide/docs/step_by_step_guide.md).

There are a large number of things that render fine in the GitHub rendering, but not the actual rendering. Are we actually using a different variant of Markdown? Or is the document rendering software we use only capable of handling a subset of Markdown? In any case, the line "Our documentation is written in Markdown language.  See [Basic writing and formatting syntax - GitHub Docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)  to get started." in [Developing Documentation](https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/3924787306/Developing+Documentation) should be updated to reflect the fact that a great number of useful features don't actually work as in GitHub...

File: docs/step_by_step_guide.md (hunk 6 above)
Comment (forsyth2) [reply docs/step_by_step_guide.md:8]: Yes that would be great. Thanks @mahf708! (I still think the docs on developing our docs should be updated to reflect that we have some differences from standard GitHub markdown.)

**Comment by rljacob** [comment]
@forsyth2 regarding rendering diffs:  the docs are converted to html from markdown by mkdocs and the html is what gets displayed on the github.io page.

**Comment by forsyth2** [comment]
> regarding rendering diffs: the docs are converted to html from markdown by mkdocs and the html is what gets displayed on the github.io page.

I'm still a little confused here. Does that mean `mkdocs` converts markdown -> html differently than how the GitHub website does it for things like READMEs and these pull request comments?

**Comment by rljacob** [comment]
Yes its a different rendering engine.   mkdocs has options to further change the rendered look and feel.  That's why the our documentation docs tell the user to use the mkdocs built-in html server to look at the result.

**Comment by rljacob** [comment]
If you want, this could be added at a higher level in the github docs:  https://e3sm-project.github.io/.  Do that, make a PR to https://github.com/E3SM-Project/E3SM-Project.github.io

**Comment by forsyth2** [comment]
@rljacob That sounds good to me. I'll do that, thanks!

**Comment by forsyth2** [comment]
Closing in favor of https://github.com/E3SM-Project/E3SM-Project.github.io/pull/2

**Pull request closed without merging**